### ISSUE TITLE
Set-ConfluenceInfo does not persist when called in a function

### ISSUE THREAD

**Issue by mcx808** [AtlassianPS/ConfluencePS#199]
I created a draft powershell script that called the following in the `begin{}` block:
`Set-ConfluenceInfo -BaseURi "$baseUri" -Credential $ConfluenceCredential`
Then in the `process{}` block I would use Get-ConfluencePage and Set-ConfluencePage with the following parameters:
`Get-ConfluencePage -SpaceKey $SpaceKey | Where-Object {$_.Title -eq $PageTitle}`

Once I was happy with the script I created a function around it and added to a module but it would always ask me for the `ApiUrl` when Get-ConfluencePage was called. 

I temporary workaround was to add the following when wrapping as a function
```
$PSDefaultParameterValues['Disabled'] = $true
$apiUri = $baseUri + '/rest/api'
Get-ConfluencePage -SpaceKey $SpaceKey -ApiUri $apiUri -Credential $ConfluenceCredential| Where-Object {$_.Title -eq $PageTitle}
```

Is there a known issue for why the behaviour is different and incompatible when called as a function instead of script?

**Comment by cwanized** [comment]
I got the same behavior.

For me, version 2.5.0 is working fine. I guess I stick with version right now...

any word from the developers?

**Comment by mcx808** [comment]
> I got the same behavior.
> 
> For me, version 2.5.0 is working fine. I guess I stick with version right now...
> 
> any word from the developers?

Nope, not a peep since I opened the issue. Considering the last commit was 3 years ago I'm guessing the project is abandoned.

**Comment by rnelson0** [comment]
In #251, Set-Info correctly sets the ApiUri value now. However, Get-Page and others still have a mandatory parameter `$ApiUri` with no default or lookup of `$script:PSDefaultParameterValues["${command}:${parameter}"]` - persistence doesn't mean much but it's technically complete.
